lib/archethic/beacon_chain/network_coordinates.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Archethic.BeaconChain.NetworkCoordinates do
+  @moduledoc """
+  Network coordinates is a way to map latency between nodes and used to determine the closest nodes
+  """
+
+  @digits ["F", "E", "D", "C", "B", "A", "9", "8", "7", "6", "5", "4", "3", "2", "1", "0"]
+
+  @type latency_patch :: {String.t(), String.t()}
+
+  @doc """
+  Compute the network patch based on the matrix latencies
+
+  The matrix must be an Hollow matrix, symmetric and all the latencies must be positive
```

Comment:
Hollow matrix can refer to many things, maybe we could rephrase is in a more formal way like this: 
"The matrix M must respect the following conditions 
- M is hollow ie M is an n × n matrix A = (aij) where aij = 0 whenever i = j (i.e. aii = 0 for all i)
- M must be symmetric and positive for all i and for all j aij = aji and aij >= 0"